iml-system-test-utils/src/lib.rs

```diff
@@ -144,6 +148,17 @@ pub enum SystemTestError {
     SystemdError(#[from] SystemdError),
 }
 
+pub trait ServerList {
+    fn to_server_list(&self) -> Vec<&str>;
+}
+
+impl<S: std::hash::BuildHasher> ServerList for HashMap<String, &[&str], S> {
+    fn to_server_list(&self) -> Vec<&str> {
+        let server_set: BTreeSet<_> = self.values().cloned().flatten().collect();
+        server_set.into_iter().map(|x| *x).collect()
+    }
+}
```

Comment:
This takes in a borrow but clones the contents.

This could be done with derefs instead:

```suggestion
impl<S: std::hash::BuildHasher> ServerList for HashMap<String, &[&str], S> {
    fn to_server_list(&self) -> Vec<&str> {        
        let server_set: BTreeSet<&str> =
            self.into_iter().flat_map(|(_, v)| *v).map(|x| *x).collect();

        server_set.into_iter().collect()
    }
}
```

Reply:
sure, i'll update.